pandas/core/reshape/tile.py

```diff
@@ -396,7 +398,12 @@ def _bins_to_cuts(
             labels = _format_labels(
                 bins, precision, right=right, include_lowest=include_lowest, dtype=dtype
             )
-        else:
+        elif labels is True:
```

Comment:
Maybe a little bit cleaner to validate `labels` immediately instead of embedding the logic here.  Right after the `duplicates` validation at the beginning of `_bins_to_cuts` you could add something like:
```python
if labels is not None and labels is not False and not is_list_like(labels):
    raise ValueError(...)
```
Or alternatively could factor out the `not`:
```python
if not (labels is None or labels is False or is_list_like(labels)):
    raise ValueError(...)
```

This would allow the rest of the logic in the `_bins_to_cut` to proceed under the assumption that `labels` is valid.  The advantage being then that this block of code can remain unchanged as the `else` block here must be list-like labels since the other two options have already been eliminated by the preceding `if` statements.

Reply:
Thanks, Ive added added similar for eliminating not None and no islike but after doing a check for is not False in as there is also an outer else block for cases where labels is False, so was a bit tricky doing them all in one go

Reply:
Thanks.  For checking all three in one go I meant moving the check further up in `_bins_to_cuts` to around line 369.